packages/eslint-plugin-turbo/lib/rules/no-undeclared-env-vars.ts

```diff
@@ -81,6 +83,10 @@ function create(context: RuleContextWithOptions): Rule.RuleListener {
     }
   });
 
+  frameworkInferencePrefixes.forEach((prefix) => {
+    regexAllowList.push(new RegExp(`^${prefix}`));
```

Comment:
This is too naive, since it doesn't account for package boundaries. It's going to auto-allow `NEXT_PUBLIC` in a Vite application.

(In practice, this probably isn't a big problem but let's see what we can do to be 100% correct...)